src/Network/Affjax/Request.purs

```diff
@@ -0,0 +1,115 @@
+module Network.Affjax.Request
+  ( Ajax()
+  , AjaxRequest()
+  , MethodName(..)
+  , Content(..)
+  , AjaxResponse()
+  , defaultRequest
+  , ajax
+  ) where
+
+import Control.Monad.Aff (Aff(), EffA(), makeAff)
+import Control.Monad.Eff (Eff())
+import Control.Monad.Eff.Exception(Error())
+import Data.Array ()
+import Data.ArrayBuffer.Types (ArrayView())
+import Data.Function (Fn8(), runFn8)
+import Data.Maybe (Maybe(..), maybe)
+import Data.Nullable (Nullable(), toNullable)
+import DOM (Document())
+import DOM.File (Blob())
+import DOM.XHR (FormData())
+import Network.HTTP (Header(..))
+
+-- | The event type for AJAX requests.
+foreign import data Ajax :: !
+
+-- | The parameters for an AJAX request.
+type AjaxRequest a =
+  { url :: String
+  , method :: MethodName
+  , headers :: [Header]
+  , content :: Maybe (Content a)
+  , username :: Maybe String
+  , password :: Maybe String
+  }
+
+-- | A HTTP method name: `GET`, `POST`, etc.
+newtype MethodName = MethodName String
```

Comment:
I wanted to avoid defining these at all originally, and just use the verbs from `purescript-http` until I saw the requirement for custom verbs. The `method` combinator does accept the `purescript-http` verbs at the moment at least.

Perhaps I should make a PR to allow custom verbs there instead.